neo/IO/Caching/DataCache.cs

```diff
@@ -119,6 +120,47 @@ public void DeleteWhere(Func<TKey, TValue, bool> predicate)
         }
 
         public IEnumerable<KeyValuePair<TKey, TValue>> Find(byte[] key_prefix = null)
+        {
+            lock (dictionary)
+            {
+                return FindUnsorted(key_prefix).OrderBy(u => u.Key, this);
```

Comment:
This forces the entire set of search results to be loaded into memory for sorting, which can lead to attacks.

Reply:
Still nowadays I have dificulties with C#.

Looking at it I would think that find it finds and returns the set, then, it sorts the results.

In this sense, maybe it needs to be split into two steps, right, @erikzhang?

Reply:
We need to do some optimization. Because the records read from the underlying storage (leveldb) is already sorted.

Reply:
Perfect, master.
We are going to reflect on that we think about an efficient way.

Reply:
I will propose another way